src/modules/didRegistry/didRegistry.service.ts

```diff
@@ -418,4 +432,56 @@ export class DidRegistry {
       })
     );
   }
+
+  /**
+   * validates update document request
+   */
+  private validDateUpdateDocumentRequest({
```

Comment:
There is a package that can help with this https://www.npmjs.com/package/ts-interface-checker. Do you think we can use it?

Reply:
At first glance it is missing : nested objects validation, enums support and conditions (which are not crucial). I'll give it a try to see if the code improves.

Reply:
it has got enums support, promising ...

Reply:
nested objects too ...